infra/cifuzz/filestore/gitlab_artifacts/__init__.py

```diff
@@ -0,0 +1,144 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Gitlab artifacts filestore implementation."""
+import logging
+
+import json
+import os
+import shutil
+import tempfile
+
+import filestore
+import http_utils
+
+# pylint:disable=no-self-use,unused-argument
+
+
+class GitlabArtifactsFilestore(filestore.BaseFilestore):
+  """Implementation of BaseFilestore using Gitlab job artifacts.
+  Either needs to use a cache.
+  Or needs a PRIVATE-TOKEN supplied to access the Gitlab API as it seems
+  the only way to know which job id to use to download an artifact
+  from another job"""
+
+  def __init__(self, config):
+    super().__init__(config)
+    # Set of downloaded artifacts
+    self.downloaded = set()
+    self.artifacts_dir = self.config.platform_conf.artifacts_dir
+    self.cache_dir = self.config.platform_conf.cache_dir
+    self.download_dir = self.config.platform_conf.download_dir
+    self.api_url = self.config.platform_conf.api_url
+
+  def _copy_from_dir(self, src, name, reason):
+    dest_dir_artifacts = os.path.join(self.config.project_src_path,
+                                      self.artifacts_dir, reason, name)
+    logging.info('Uploading %s to artifacts to %s.', reason, dest_dir_artifacts)
+    shutil.copytree(src, dest_dir_artifacts)
+    # Saves to gitlab cache if any is specified.
+    if self.cache_dir:
```

Comment:
Why would one not be specified?
What purpose does allowing a user not to specify a cache serve?

Reply:
If you want to use a gitlab cache for ClusterFuzzLite, this cache needs to be shared among your gitlab runners.
If you do not want to bother to setup this shared cache, you can still get the filestore functionality, but then you need to resort to artifacts, and thus you need to let your CI use a token